src/Microsoft.PowerShell.Security/security/CertificateProvider.cs

```diff
@@ -3340,17 +3340,25 @@ private DnsNameRepresentation GetDnsNameRepresentation(string dnsName)
         /// </summary>
         public DnsNameProperty(X509Certificate2 cert)
         {
-            _dnsList = new List<DnsNameRepresentation>();
-
-            // extract DNS name from subject distinguish name
-            // if it exists and does not contain a comma
-            // a comma, indicates it is not a DNS name
-            if (cert.Subject.StartsWith(distinguishedNamePrefix, StringComparison.OrdinalIgnoreCase) &&
-                !cert.Subject.Contains(','))
+            // Extract DNS name from Subject distinguished name
+            foreach (X500RelativeDistinguishedName rdn in cert.SubjectName.EnumerateRelativeDistinguishedNames())
             {
-                string parsedSubjectDistinguishedDnsName = cert.Subject.Substring(distinguishedNamePrefix.Length);
-                DnsNameRepresentation dnsName = GetDnsNameRepresentation(parsedSubjectDistinguishedDnsName);
-                _dnsList.Add(dnsName);
+                // Check distinguished name has single element and has Common Name (CN) OID
+                // We skip multi value CN-ID RDNs since in RFC it is not clear how they are handled
+                // Given we have cases where Subject field is duplicated in SAN, we continue to add entries to DNS name list
+                if (!rdn.HasMultipleElements)
+                {
+                    string oid = rdn.GetSingleElementType().Value;
+                    string rdnValue = rdn.GetSingleElementValue();
+
+                    if (!string.IsNullOrEmpty(oid) &&
+                        !string.IsNullOrEmpty(rdnValue) &&
+                        oid.Equals(CommonNameOid.Value, StringComparison.OrdinalIgnoreCase))
```

Comment:
Oh, my comment (about Oid type) was overkill. I thought the type has specific Equals() method/operator but looking .Net code I see there is used simple string comparison or Oid-s.  I think it is better to revert the change.

Reply:
And no need to use OrdinalIgnoreCase (only numbers and dots is in the string) and do extra null checks if CommonNameOid will be a string _const_:
```c#
CommonNameOid.Equals(rdnValue, StringComparison.Ordinal)
```

Reply:
Yeah there was no `Equals` override which would have made this cleaner and straightforward. I will revert the change and apply other suggestion.

Reply:
@iSazonov reverted and changed to Ordinal comparison in 64b93a8d9b02ffbe1d9260421095a4c8b1b2e0e2. I think the existing null checks for OID and RDN value should be fine. Probably don't need one for `CommonNameOid` since its just a const.